src/test/java/net/sf/jsqlparser/statement/select/HiveTest.java

```diff
@@ -46,4 +46,24 @@ public void testLeftSemiJoin() throws Exception {
 
         assertSqlCanBeParsedAndDeparsed(sql, true);
     }
+
+    @Test
+    public void testGroupByGroupingSets() throws Exception {
+        String sql;
+        Statement statement;
+
+        sql = "SELECT\n"
+            + "    C1, C2, C3, MAX(Value)\n"
+            + "FROM\n"
+            + "    Sometable\n"
+            + "GROUP BY C1, C2, C3 GROUPING SETS ((C1, C2), (C1, C2, C3), ())";
+
+        statement = CCJSqlParserUtil.parse(sql);
```

Comment:
This hole parsing process is done using  assertSqlCanBeParsedAndDeparsed. So you could skip it.